pybamm/quick_plot.py

```diff
@@ -0,0 +1,227 @@
+        self.electrolyte_concentration, = plt.plot(self.x, self.c_e(t, self.x), lw=2)
+        plt.axis(self.axis["Electrolyte concentration"])
+
+        plt.subplot(243)
+        plt.xlabel("x_p")
+        plt.ylabel("Positive particle surface concentration")
+        self.positive_particle_concentration, = plt.plot(
+            self.x_p, self.c_s_p_surf(t, self.x_p), lw=2
+        )
+        plt.axis(self.axis["Positive particle surface concentration"])
+
+        plt.subplot(244)
+        plt.xlabel("Time")
+        plt.ylabel("Total current density")
+        self.current, = plt.plot(self.t, self.i_cell(self.t), lw=2)
+        self.current_point, = plt.plot(
+            [t], [self.i_cell(t)], marker="o", markersize=5, color="red"
+        )
+        plt.axis(self.axis["Total current density"])
+
+        # negative electrode potential
+        plt.subplot(245)
+        plt.xlabel("x")
+        plt.ylabel("Negative electrode potential [V]")
+        self.negative_electrode_potential, = plt.plot(
+            self.x_n, self.phi_s_n(t, self.x_n), lw=2
+        )
+        plt.axis(self.axis["Negative electrode potential [V]"])
+
+        # electrolyte potential
+        plt.subplot(246)
+        plt.xlabel("x")
+        plt.ylabel("Electrolyte potential [V]")
+        self.electrolyte_potential, = plt.plot(self.x, self.phi_e(t, self.x), lw=2)
+        plt.axis(self.axis["Electrolyte potential [V]"])
+
+        # positive electrode potential
+        plt.subplot(247)
+        plt.xlabel("x")
+        plt.ylabel("Positive electrode potential [V]")
+        self.positive_electrode_potential, = plt.plot(
+            self.x_p, self.phi_s_p(t, self.x_p), lw=2
+        )
+        plt.axis(self.axis["Positive electrode potential [V]"])
+
+        # voltage
+        plt.subplot(248)
+        plt.xlabel("Time")
+        plt.ylabel("Terminal voltage [V]")
+        self.voltage, = plt.plot(self.t, self.V(self.t), lw=2)
+        self.voltage_point, = plt.plot(
+            [t], [self.V(t)], marker="o", markersize=5, color="red"
+        )
+        plt.axis(self.axis["Terminal voltage [V]"])
+
+    def dynamic_plot(self, testing=False):
+        """
+        Generate a dynamic plot with a slider to control the time. We recommend using
+        ipywidgets instead of this function if you are using jupyter notebooks
+        """
+
+        # create an initial plot at time 0
+        self.plot(0)
+
+        axcolor = "lightgoldenrodyellow"
+        axfreq = plt.axes([0.315, 0.05, 0.37, 0.03], facecolor=axcolor)
+        self.sfreq = Slider(axfreq, "Time", 0, self.t.max(), valinit=0)
+        self.sfreq.on_changed(self.update)
+
+        plt.subplots_adjust(
+            top=0.92, bottom=0.15, left=0.10, right=0.9, hspace=0.5, wspace=0.5
+        )
+
+        if not testing:
+            plt.show()
+
+    def update(self, val):
+        """
+        Update the plot in self.plot() with values at new time
+        """
```

Comment:
from the codecov it looks like this isn't being tested, even though you have a test for it? not sure why